File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        IEnumerable<int> digits = Convert(number);
        return number == digits.Select(x => (int)Math.Pow(x, digits.Count())).Sum();

        // Alternate solution that doesn't need the Convert function but needs lots of type conversion
        // string numberStr = number.ToString();
        // return number == numberStr.Select(x => (int)Math.Pow(int.Parse(x.ToString()), numberStr.Length)).Sum();
    }

    private static IEnumerable<int> Convert(int number) {
        List<int> result = new List<int>();

        while (number > 0) {
            result.Add(number % 10);
            number /= 10;
        }

        return result;
    }
}

File: erlang/armstrong-numbers/src/armstrong_numbers.erl
-module(armstrong_numbers).

-export([is_armstrong_number/1]).


is_armstrong_number(Number) -> Length = length(integer_to_list(Number)), 
                               trunc(calculate(Number, Length)) =:= Number.

calculate(0, _) -> 0;
calculate(Number, Power) -> math:pow(Number rem 10, Power) + calculate(Number div 10, Power).
